Modules/Module_Processing/Module_Population/xASL_stat_GetDICOMStatistics.m

```diff
@@ -36,78 +39,55 @@ function xASL_stat_GetDICOMStatistics(x, ScanType, HasSessions, bOverwrite)
     HasSessions = false;
 end
 
-% Mat fields that will be printed (some are in x.(field) and some in x.Q.(field))
-matFields = {'RepetitionTime' 'EchoTime' 'NumberOfTemporalPositions' 'MRScaleSlope' 'RescaleSlopeOriginal' 'RescaleIntercept' 'Scanner' 'SliceReadoutTime'};
-nFields = length(matFields);
 PathTSV = fullfile(x.D.DICOMparameterDir, ['QuantificationParameters_' ScanType '.tsv']);
 
 % Print header
 TSV = {'participant_id' 'session'};
-TSV(1,3:2+nFields) = matFields(1:nFields);
 
-% Initialize empty table
-numElements = size(TSV,2);
-numSubjectsSessions = x.dataset.nSubjects*x.dataset.nSessions;
-for iSubjSess=1:numSubjectsSessions
-    TSV(1+iSubjSess,:) = repmat({' '},1, numElements);
-end
 
-%% 1) Load & save individual parameter files
-fprintf('%s\n','Loading & saving individual parameter files...  ');
+%% 1. Loading JSON sidecars
+fprintf('%s\n', ['Loading ' ScanType '.json sidecars...  ']);
 
 for iSubject=1:x.dataset.nSubjects
     for iSession=1:x.dataset.nSessions
 
         % Track progress
         iSubjSess = (iSubject-1)*x.dataset.nSessions+iSession;
-        xASL_TrackProgress(iSubjSess, x.dataset.nSubjects * x.dataset.nSessions);        
+        xASL_TrackProgress(iSubjSess, x.dataset.nSubjects * x.dataset.nSessions);
         
         % Default parameters
         TSV{1+iSubjSess,1} = x.SUBJECTS{iSubject};
         TSV{1+iSubjSess,2} = x.SESSIONS{iSession};
         
-        TSV(1+iSubjSess, 3:nFields+2) = repmat({NaN}, [1 nFields]);
-        x.S.par(iSubjSess,1:nFields) = NaN;
+        % TSV(1+iSubjSess, 3:nFields+2) = repmat({NaN}, [1 nFields]);
+        % x.S.par(iSubjSess,1:nFields) = NaN;
         
         % Define paths
 		if HasSessions
-            PathMAT = fullfile(x.dir.xASLDerivatives, x.SUBJECTS{iSubject}, x.SESSIONS{iSession}, [ScanType '_Parms.mat']);% Legacy
             PathJSON = fullfile(x.dir.xASLDerivatives, x.SUBJECTS{iSubject}, x.SESSIONS{iSession}, [ScanType '.json']);
         else
-            PathMAT = fullfile(x.dir.xASLDerivatives, x.SUBJECTS{iSubject}, [ScanType '_Parms.mat']); % Legacy
             PathJSON = fullfile(x.dir.xASLDerivatives, x.SUBJECTS{iSubject}, [ScanType '.json']); 
 		end
-		if exist(PathMAT,'file') || exist(PathJSON,'file')
-			Parms = xASL_adm_LoadParms(PathMAT,[],0);
-			% Parms = xASL_io_ReadJson(PathJSON);
+		if exist(PathJSON, 'file')
+			Parms = xASL_io_ReadJson(PathJSON);
```

Comment:
Careful about BIDS vs Legacy.
adm_LoadParms either loads MAT or JSON.

So practically, the old version of code already was reading JSONs most of the time. But that's not a problem, I agree to remove the MAT backwards compatibility. 
They issue is that loadParms function was converting JSONs from BIDS to Legacy and now we are reading JSONs direct.

So we should add this code:
    % Convert parameters to BIDS
	Parms = xASL_bids_parms2BIDS([], Parms, 0, 1);

As a second thing - you then have to be careful that you'll have the fields under Parms.Q -> as was in the previous code.

However, if you say that this is intentional to use BIDS and not Legacy fields, then I'm fine as well -> I just wanted to inform you about this.

Reply:
I understand what you mean.

This function just loads json-files to take parameters from it that it will move (without changing, except from putting them to a string) to a table. This table is then printed as TSV. So it is intentional that we keep all in BIDS and don't go to legacy.